src/Diagnostics.DataProviders/DataProviders/SupportObserverDataProvider.cs

```diff
@@ -0,0 +1,151 @@
+            var slotTimeRangeCaseSensitiveDictionary = JsonConvert.DeserializeObject<Dictionary<string, List<RuntimeSitenameTimeRange>>>(response);
+            var slotTimeRange = new Dictionary<string, List<RuntimeSitenameTimeRange>>(slotTimeRangeCaseSensitiveDictionary, StringComparer.CurrentCultureIgnoreCase);
+            return slotTimeRange;
+        }
+
+        public Task<IEnumerable<Dictionary<string, string>>> GetServerFarmsInResourceGroup(string subscriptionName, string resourceGroupName)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<string> GetServerFarmWebspaceName(string subscriptionId, string serverFarm)
+        {
+            throw new NotImplementedException();
+        }
+
+        public async Task<IEnumerable<string>> GetSiteHostNames(string siteName)
+        {
+            var response = await GetObserverResource($"sites/{siteName}/hostnames?api-version=2.0");
+            var hostnames = response.Remove(0, 1).Remove(response.Length - 2, 1).Split(new char[] { ',' });
+            return hostnames;
+        }
+
+        public Task<string> GetSiteResourceGroupName(string siteName)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<IEnumerable<Dictionary<string, string>>> GetSitesInResourceGroup(string subscriptionName, string resourceGroupName)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<IEnumerable<Dictionary<string, string>>> GetSitesInServerFarm(string subscriptionId, string serverFarmName)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<string> GetSiteWebSpaceName(string subscriptionId, string siteName)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<string> GetStorageVolumeForSite(string stampName, string siteName)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<string> GetWebspaceResourceGroupName(string subscriptionId, string webSpaceName)
+        {
+            throw new NotImplementedException();
+        }
+
+        public async Task<dynamic> GetSite(string siteName)
+        {
+            var response = await GetObserverResource($"sites/{siteName}");
+            var siteObject = JsonConvert.DeserializeObject(response);
+            return siteObject;
+        }
+
+        public async Task<dynamic> GetSite(string stampName, string siteName)
+        {
+            return await GetSite(siteName);
+        }
+
+        private async Task<string> GetObserverResource(string url)
+        {
+            var request = new HttpRequestMessage(HttpMethod.Get, url);
+            request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", await GetAccessToken());
+            var response = await _httpClient.SendAsync(request);
+            response.EnsureSuccessStatusCode();
```

Comment:
How is the exception thrown here handled at caller side. Should we log an event indicating DataSource Errored out.

Right now, there is no ETW event in Data providers project as we recently added ETW for main API calls.
As a separate work-item, It will be nice to add few common events like summary event about latency and exception to some base class so that every data provider gets it automatically.

Reply:
The exception here is going to be tracked by the ETW events issued at the API level. The diagnostic logger middleware that you created should capture the exception details and show the stack trace in our kusto logs so that we can see where the exception is thrown from.

Yes I agree about the summary event for data providers. I added an [issue](https://github.com/ShekharGupta1988/AppLensV2/issues/1061) for this

Reply:
oh cool.